### ISSUE TITLE
Crash using bicycle profile on `master`

### ISSUE THREAD

**Issue by TheMarex** [Project-OSRM/osrm-backend#4129]
While doing QA on testing 5.8.0 I found the following crashing request:

```
$1 = (const osrm::engine::api::RouteParameters &) @0x32e6e30: {<osrm::engine::api::BaseParameters> = {coordinates = std::vector of length 2, capacity 2 = {{lon = {__value = -95727800}, lat = {
          __value = 39044000}}, {lon = {__value = -111935242}, lat = {__value = 33418938}}}, hints = std::vector of length 0, capacity 0, radiuses = std::vector of length 0, capacity 0, 
    bearings = std::vector of length 0, capacity 0, approaches = std::vector of length 0, capacity 0, generate_hints = true}, steps = true, alternatives = false, annotations = false, 
  annotations_type = osrm::engine::api::RouteParameters::None, geometries = osrm::engine::api::RouteParameters::GeoJSON, overview = osrm::engine::api::RouteParameters::Full, 
  continue_straight = {<boost::optional_detail::optional_base<bool>> = {<boost::optional_detail::optional_tag> = {<No data fields>}, m_initialized = false, m_storage = {dummy_ = {data = "", 
          aligner_ = 0 '\000'}}}, <No data fields>}}
```

The full backtrace I could pull can be found [here](https://gist.github.com/TheMarex/bf8abafc734a13a052e76cae3a99ef13)

**Comment by oxidase** [comment]
@TheMarex query to reproduce `localhost:5000/route/v1/driving/-111.93991184234619,33.422102120361295;-111.935218,33.418991?overview=false&steps=true`

**Comment by TheMarex** [comment]
Fixed by #4130
